message-editing: Add support for `Copy and close` button.

akshatdalton opened the pull request:
This commit adds support for the `Copy and close` button in textarea when message editing is still enabled. Earlier a `Copy and close` button only appears if message editing time has passed. But now even though a message is editable or not, the `Copy and close` button would appear.

This will be particularly helpful for the users who want to copy their own message content and their realm setting: `Allow message editing` is set to: `Any time`.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->

https://chat.zulip.org/#narrow/stream/6-frontend/topic/Add.20copy.20and.20close.20button

**Testing plan:** <!-- How have you tested? -->

Tested locally with all possible settings.



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

Signior-X commented:
I ran this on my local machine. It works fine. Just some of my observations
1. If I edit a message and then click `copy and close` button, it doesn't saves the content and closes the view.

I think there can be two possibilities, either remove the close from copy button here, or secondly we should also save the edited content before saving. Let's ask on CZO first.

timabbott commented:
This merge conflicts with an adjance tPR that we just merged: f872c95c657a8a3c18415b88d6cf212009ac8684

akshatdalton commented:
I have updated my PR along with the commit message.
This change will only affect the message editing. Added `Copy` button would not close the compose box as it should not confuse the users that their edited content is lost or saved unintentionally.
The `Copy` button would copy the present content in the compose box.
